src/main/scala/beam/sim/BeamHelper.scala

```diff
@@ -576,13 +576,13 @@ trait BeamHelper extends LazyLogging {
         fileFormat match {
           case "csv" =>
             val beamScenario = loadScenario(beamConfig)
-            val emptyScenario = ScenarioBuilder(matsimConfig).withNetwork(beamScenario.network).build
             val scenario = {
               val source = new BeamScenarioSource(
                 beamConfig,
                 rdr = readers.BeamCsvScenarioReader
               )
-              new BeamScenarioLoader(emptyScenario, beamScenario, source, new GeoUtilsImpl(beamConfig)).loadScenario()
+              val scenarioBuilder = ScenarioBuilder(matsimConfig, beamScenario.network)
+              new BeamScenarioLoader(scenarioBuilder, beamScenario, source, new GeoUtilsImpl(beamConfig)).loadScenario()
             }.asInstanceOf[MutableScenario]
```

Comment:
Won't it fail with `ClassCastException`  because of casting `BeamScenarioLoader` to `MutableScenario`?

Reply:
No. there is a method call `.loadScenario()` at the end